streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java

```diff
@@ -64,4 +70,70 @@ public void testReprocessingFromScratchAfterResetWithIntermediateUserTopic() thr
     public void testReprocessingFromScratchAfterResetWithoutIntermediateUserTopic() throws Exception {
         super.testReprocessingFromScratchAfterResetWithoutIntermediateUserTopic();
     }
+
+    @Test
+    public void testReprocessingFromLatestEarliestAfterResetWithoutIntermediateUserTopic() throws Exception {
+        super.testReprocessingFromLatestAfterResetWithoutIntermediateUserTopic();
+    }
+
+    @Test
+    public void testReprocessingFromOffsetAfterResetWithoutIntermediateUserTopic() throws Exception {
+        super.testReprocessingFromOffsetAfterResetWithoutIntermediateUserTopic();
+    }
+
+    @Test
+    public void testReprocessingByShiftPositiveAfterResetWithoutIntermediateUserTopic() throws Exception {
+        super.testReprocessingByShiftPositiveAfterResetWithoutIntermediateUserTopic();
+    }
+
+    @Test
+    public void testReprocessingByShiftNegativeAfterResetWithoutIntermediateUserTopic() throws Exception {
+        super.testReprocessingByShiftNegativeAfterResetWithoutIntermediateUserTopic();
+    }
+
+    @Test
+    public void testReprocessingFromFileAfterResetWithoutIntermediateUserTopic() throws Exception {
+        super.testReprocessingFromFileAfterResetWithoutIntermediateUserTopic();
+    }
+
+    @Test
+    public void testReprocessingFromDateTimeAfterResetWithoutIntermediateUserTopic() throws Exception {
+        super.testReprocessingFromDateTimeAfterResetWithoutIntermediateUserTopic();
+    }
+
+    @Test
+    public void testReprocessingByDurationAfterResetWithoutIntermediateUserTopic() throws Exception {
+        super.testReprocessingByDurationAfterResetWithoutIntermediateUserTopic();
+    }
+
+    @Test
+    public void testDateTimeFormats() throws ParseException {
+        //check valid formats
+        invokeGetDateTimeMethod(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS"));
+        invokeGetDateTimeMethod(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ"));
+        invokeGetDateTimeMethod(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX"));
+        invokeGetDateTimeMethod(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXX"));
+        invokeGetDateTimeMethod(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX"));
+
+        //check some invalid formats
+        try {
+            invokeGetDateTimeMethod(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss"));
+            fail("Call to getDateTime should fail");
+        } catch (Exception e) {
```

Comment:
nit: add `final` (same below)

Reply:
Not sure where the final should be. Maybe you're referring to the next comment?

Reply:
`} catch (final Exception e) {`